src/net/sourceforge/kolmafia/KoLAdventure.java

```diff
@@ -678,6 +709,11 @@ private void validate1() {
     this.isValidAdventure = true;
   }
 
+  public boolean isAccessible() {
+    this.validate1();
```

Comment:
could `isAccessible` contain the logic that `validate1` currently contains (but instead of setting `this.isValidAdventure`, just returning a boolean)? And then `validate1` calls `this.isValidAdventure = this.isAccessible()`?

As-is there's a potential race here, and I think the other way is neater anyway.

Reply:
I will look into this. It does seem cleaner.

Reply:
I did this. It also makes the code more compact.
There are a couple of places I am suspicious of the current behavior.
Why not simply say that the Airship is accessible if you have given given the quest?
Yes, you have to plant a beanstalk first, but the code is intentionally "giving the between battle script the opportunity to get a bean".

I don't think this PR is the place to think about changing the validation. Just refactoring to be cleaner.